CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/recipes/MCRecipeManager.java

```diff
@@ -175,6 +182,23 @@ public void addShapeless(String name, IItemStack output, IIngredient[] ingredien
         }
         recipesToAdd.add(new ActionAddShapelessRecipe(name, output, ingredients, function, action, false));
     }
+
+    @Override
+    public void addShapeless(String group, String name, IItemStack output, IIngredient[] ingredients, @Optional IRecipeFunction function, @Optional IRecipeAction action) {
+        boolean valid = output != null;
+        for(IIngredient ing : ingredients) {
+            if(ing == null) {
+                valid = false;
```

Comment:
Putting a break here would save some CPU time.

Reply:
That's a duplication of the current addShapeless command as there was no easier way to access the recipe object.